doc/History.md

```diff
@@ -0,0 +1,48 @@
+## MongooseIM history
+
+### Fork of ejabberd
+
+MongooseIM's birthplace is a private Erlang Solutions' branch of ProcessOne's ejabberd - an XMPP/Jabber server written in Erlang.
+What would later become a leading, highly customisable and scalable XMPP platform, originated in a single idea - storing all internal strings in binaries instead of lists.
```

Comment:
It was a bit more complex than that. "Binarification" was just one of the ideas. I wouldn't name it the only, single improvement we had in mind.